config/routes.rb

```diff
@@ -16,14 +16,14 @@
   # We just need to define a "new" view to prompt for user name,
   # and password.
   get    "/login",   to: "sessions#new"
-  post   "/login",   to: "sessions#create"
   delete "/logout",  to: "sessions#destroy"
   post   "/logout",  to: "main#logout"
 
   # Routes for Google authentication, note that these need to be
   # here for ominauth middleware whose route is /auth/google_oauth2,
   # which is not specified in this file, (because the middleware did it).
-  get "auth/:provider/callback", to: "sessions#googleAuth"
+  get "auth/google_oauth2/callback", to: "sessions#googleAuth"
+  get "auth/microsoft_graph/callback", to: "sessions#microsoftAuth"
```

Comment:
In the og code, `:provider` was used.
I wonder if we can use the same idea, so here we have something like
`get "auth/:provider/callback", to: "sessions#auth"`
And in `session#auth` call `generalAuth(:provider)`. Im not sure where this `:provider` var is set, but if we can use it in `sessionController#auth`, it would make adding more providers more scalable.

Reply:
Done!